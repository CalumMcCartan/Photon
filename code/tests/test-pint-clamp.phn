func int main()
{
    pint x;
    int i;

    x = 42;
    print(x);
    x = 255;
    print(x);  
    x = 256;
    print(x);
    x = 1000;
    print(x); 
    x = 0;
    print(x);  
    x = -1;
    print(x);

    i = 10;
    x = 1 + 1;
    print(x); 

    pint x2;
    x = 200;
    x2 = 200;
    print(x + x2);

    x = getBigInt();
    print(x);

    x = x2;
    printb(x == x2);

    printPint(66);
    printPint(-12);

    return 0;
}

func int getBigInt() {
    int z;
    z = 500;
    return z;
}

func int printPint(pint p) {
    print(p);
    return p;
}
