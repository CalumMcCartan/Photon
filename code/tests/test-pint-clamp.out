42
255
255
255
0
0
2
255
255
1
66
0